app/src/main/java/com/example/wrk/CreateAdapter.java

```diff
@@ -74,42 +77,46 @@ public ViewHolder(View itemView, final Context context) {
             ibEdit = itemView.findViewById(R.id.ibEdit);
         }
 
+        @RequiresApi(api = Build.VERSION_CODES.N)
         public void bind(WorkoutTemplate template) throws JSONException, ParseException {
-            tvTemplateTitle.setText(template.getTitle());
-            queryComponents(template);
-
-            bodyParts = new ArrayList<>();
-            // store body parts included in workout in a list
-            for (int i = 0; i < workoutComponents.size(); i++) {
-                String body = workoutComponents.get(i).getExercise().getBodyPart();
-                bodyParts.add(body);
-            }
-
-            // remove any repetitions of body parts
-            Set<String> set = new HashSet<>(bodyParts);
-            bodyParts.clear();
-            bodyParts.addAll(set);
-            tvBodyParts.append(String.join(", ", bodyParts));
-
-            ibEdit.setOnClickListener(new View.OnClickListener() {
-                @Override
-                public void onClick(View v) {
-                    Intent i = new Intent(mContext, ScratchCreateActivity.class);
-                    i.putExtra("template", Parcels.wrap(template));
-                    mContext.startActivity(i);
-                }
-            });
-            ibStart.setOnClickListener(new View.OnClickListener() {
-                @Override
-                public void onClick(View v) {
-                    Intent i = new Intent(mContext, TrackerActivity.class);
-                    i.putExtra("template", Parcels.wrap(template));
-                    mContext.startActivity(i);
+            if (template.isSavedByCurrentUser()) {
+                tvTemplateTitle.setText(template.getTitle());
+                queryComponents(template);
+
+                bodyParts = new ArrayList<>();
+                // store body parts included in workout in a list
+                for (int i = 0; i < workoutComponents.size(); i++) {
+                    String body = workoutComponents.get(i).getExercise().getBodyPart();
+                    bodyParts.add(body);
                 }
-            });
+
+                // remove any repetitions of body parts
+                Set<String> set = new HashSet<>(bodyParts);
+                bodyParts.clear();
+                bodyParts.addAll(set);
```

Comment:
I feel it would be cleaner to add elements to a set in the for loop above and then cast the set to array. Or I think depending on Java version you can directly call String.join on set.

Reply:
Good idea. I will make that change.